### PR TITLE
Fix const_relevance and default qvar handler with sort polymorphism 

### LINKED ISSUE #18339: Anomaly "File "kernel/constr.ml", line 1483, characters 28-34: Assertion failed." with sort poly
~~~coq
  Inductive Box@{s|u|} (A:Type@{s|u}) := box (_:A).

  Definition t1@{s|u|} (A:Type@{s|u}) (x y : A) := box _ x.
  Definition t2@{s|u|} (A:Type@{s|u}) (x y : A) := box _ y.

  Definition ignore@{s|u|} {A:Type@{s|u}} (x:A) := tt.
  Definition unfold_ignore@{s|u|} (A:Type@{s|u}) : ignore (t1 A) = ignore (t2 A) := eq_refl.
~~~
The kernel first tries first conversion `t1 A == t2 A` so looks at the relevance of  `t1`, it's a (bound) qvar so assert false in the kernel default_evar_handler.

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #18339

### CODE REVIEW

File: kernel/cClosure.ml [hunk 1]
```diff
@@ -309,7 +309,10 @@ let is_irrelevant info r = match info.i_cache.i_mode with
 | Reduction -> false
 | Conversion -> match r with
   | Sorts.Irrelevant -> true
-  | Sorts.RelevanceVar q -> not (info.i_cache.i_sigma.qvar_relevant q)
+  | Sorts.RelevanceVar q ->
+    (* env_qualities are bound qualities so must be treated as relevant *)
+    not (Sorts.QVar.Set.mem q info.i_cache.i_env.env_qualities)
+    && not (info.i_cache.i_sigma.qvar_relevant q)
```
Comment (ppedrot) [inline kernel/cClosure.ml:315]: This should be the responsibility of qvar_relevant to perform this check so as to centralize the code handling qualities.

**Review by ppedrot — APPROVED** [review]

**Comment by herbelin** [comment]
Might be useful for #17950 (7d14b178b31b), thanks!

**Comment by ppedrot** [comment]
@SkySkimmer the lean importer now needs an overlay.

**Comment by SkySkimmer** [comment]
I reverted the API change instead of doing an overlay
(it was just to be able to do `let create env ?(evars=of_env env) flags = ...` instead of `let create ?evars flags env = let evars = Option.default (of_env env) evars in ...`, not worth it)

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**